lightning/src/ln/features.rs

```diff
@@ -160,29 +281,15 @@ impl NodeFeatures {
 	/// relevant in a node-context features and creates a node-context features from them.
 	/// Be sure to blank out features that are unknown to us.
 	pub(crate) fn with_known_relevant_init_flags(init_ctx: &InitFeatures) -> Self {
+		let byte_count = T::KNOWN_FEATURE_MASK.len();
 		let mut flags = Vec::new();
-		for (i, feature_byte)in init_ctx.flags.iter().enumerate() {
-			match i {
-				// Blank out initial_routing_sync (feature bits 2/3), gossip_queries (6/7),
-				// gossip_queries_ex (10/11), option_static_remotekey (12/13), and
-				// payment_secret (14/15)
-				0 => flags.push(feature_byte & 0b00110011),
-				1 => flags.push(feature_byte & 0b00000011),
-				_ => (),
+		for (i, feature_byte) in init_ctx.flags.iter().enumerate() {
+			if i < byte_count {
+				flags.push(feature_byte & T::KNOWN_FEATURE_MASK[i]);
```

Comment:
This probably needs T::KNOWN_FEATURE_MASK & Init::KNOWN_FEATURE_MASK, I think.

Reply:
What I have should be equivalent to the previous implementation. Are you saying that the behavior isn't correct?

Reply:
I think its correct with today's features, but if there is a bit set in the init message that we don't think makes sense in an init context, we probably shouldn't be copying it blindly to another context.